src/context.ts

```diff
@@ -129,6 +135,46 @@ export class Context {
     return await this.listTabsMarkdown();
   }
 
+  private async _initializeSessionFile() {
+    if (!this.config.saveSession)
+      return;
+
+    const timestamp = new Date().toISOString();
+    const fileName = `session${timestamp}.yml`;
+    this._sessionFile = await outputFile(this.config, fileName);
+
+    // Initialize empty session file
+    await fs.promises.writeFile(this._sessionFile, '# Session log started at ' + timestamp + '\n', 'utf8');
+  }
+
+  private async _logSessionEntry(toolName: string, params: Record<string, unknown>, snapshotFile?: string) {
+    if (!this.config.saveSession || !this._sessionFile)
```

Comment:
This is a bug, if outputFile takes time and this._sessionFile is assigned late, this will return early.